assets/src/stories-editor/components/editor-carousel/reordererIndicator.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/compose';
+import { Component } from '@wordpress/element';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { Button, Draggable, DropZone, Tooltip } from '@wordpress/components';
+
+/**
+ * Parses drag & drop events to ensure the event contains valid transfer data.
+ *
+ * @param {Object} event
+ * @return {Object} Parsed event data.
+ */
+const parseDropEvent = ( event ) => {
+	let result = {
+		srcRootClientId: null,
+		srcClientId: null,
+		srcIndex: null,
+		type: null,
+	};
+
+	if ( ! event.dataTransfer ) {
+		return result;
+	}
+
+	try {
+		result = Object.assign( result, JSON.parse( event.dataTransfer.getData( 'text' ) ) );
+	} catch ( err ) {
+		return result;
+	}
+
+	return result;
+};
+
+/**
+ * A single item within the list of pages to be reordered.
+ *
+ * Re-uses existing Draggable and DropZone provided by WordPress
+ * in order to not re-invent the wheel.
+ */
+class ReordererIndicator extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			isDragging: false,
+		};
+
+		this.onDrop = this.onDrop.bind( this );
+	}
+
+	getInsertIndex( position ) {
+		const { index } = this.props;
+
+		if ( index !== undefined ) {
+			return position.x === 'right' ? index + 1 : index;
+		}
+	}
+
+	onDrop( event, position ) {
+		const { page: { clientId }, movePageToPosition, index } = this.props;
+		const { srcClientId, srcIndex, type } = parseDropEvent( event );
+
+		const isBlockDropType = ( dropType ) => dropType === 'block';
+		const isSameBlock = ( src, dst ) => src === dst;
+
+		if ( ! isBlockDropType( type ) || isSameBlock( srcClientId, clientId ) ) {
+			return;
+		}
+
+		const positionIndex = this.getInsertIndex( position );
+		const insertIndex = srcIndex < index ? positionIndex - 1 : positionIndex;
+		movePageToPosition( srcClientId, insertIndex );
+	}
+
+	componentDidMount() {
+		this.props.setOrder();
+	}
+
+	render() {
+		const { page, index, currentPage, onClick } = this.props;
+		const { clientId } = page;
+		const pageElementId = `reorder-page-${ clientId }`;
+		const transferData = {
+			type: 'block',
+			srcIndex: index,
+			srcClientId: clientId,
+		};
+
+		/* translators: %s: Page number */
+		const label = ( pageNumber ) => sprintf( __( 'Page %s', 'amp' ), pageNumber );
+
+		/* translators: %s: Page number */
+		const toolTip = ( pageNumber ) => sprintf( __( 'Go to page %s', 'amp' ), pageNumber );
+		const className = page.clientId === currentPage ? 'amp-story-editor-carousel-item amp-story-editor-carousel-item--active' : 'amp-story-editor-carousel-item';
+
+		return (
+			<div className="amp-story-reorderer-carousel-item">
```

Comment:
What's the reason for not using `<ul>` and `<li>` anymore? Was something not working?

Reply:
@swissspidy `DropZone` component by default generates `div` and we are outputting that just before every indicator item in `indicator.js`. `li` and `div` within `ul` is not a good DOM structure, so move to use only `div`.

Reply:
What is bad about

```html
<ul>
	<li>
		<div> ... </div>
	</li>
</ul>
```

That is more semantic than just using divs everywhere

Reply:
@swissspidy You are right. I thought that `li` can not hold `div` and bump out it. I revert back to the same DOM (`ul-li`) structure with drag and drop functionality.

![image](https://user-images.githubusercontent.com/1784821/62040139-a2b05300-b216-11e9-955a-aabc049b32b9.png)

ref: https://stackoverflow.com/questions/6449772/can-i-use-a-div-inside-a-list-item